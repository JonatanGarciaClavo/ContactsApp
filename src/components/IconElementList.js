import React from 'react';
import { IconMenu, MenuItem, IconButton } from 'material-ui';
import MoreVertIcon from 'material-ui/lib/svg-icons/navigation/more-vert';
import { LIST_MODE, CARD_MODE } from '../constants/contact-list-mode-constants';

const IconElementList = ({ changeListMode }) => (
  <IconMenu
    iconButtonElement={
      <IconButton><MoreVertIcon /></IconButton>
    }
    targetOrigin={{ horizontal: 'right', vertical: 'top' }}
    anchorOrigin={{ horizontal: 'right', vertical: 'top' }}
  >
    <MenuItem primaryText="List" onTouchTap={() => changeListMode(LIST_MODE)} />
    <MenuItem primaryText="Card" onTouchTap={() => changeListMode(CARD_MODE)} />
  </IconMenu>
);

IconElementList.propTypes = {
  changeListMode: React.PropTypes.func.isRequired,
};

export default IconElementList;
