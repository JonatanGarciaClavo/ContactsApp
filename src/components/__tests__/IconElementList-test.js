import React from 'react';
import renderer from 'react-test-renderer';
import IconElementList from '../IconElementList';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

// https://github.com/facebook/jest/issues/1353
jest.mock('react/lib/ReactDefaultInjection');

describe('Testing IconElementList component', () => {
  it('renders IconElementList using Snapshots', () => {
    expect(renderer.create(
    	<MuiThemeProvider>
	      <IconElementList
	        changeListMode={jest.fn}
	      />
	     </MuiThemeProvider>
    )).toMatchSnapshot();
  });
});
