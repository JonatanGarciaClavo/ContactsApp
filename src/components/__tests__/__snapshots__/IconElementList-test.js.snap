exports[`Testing IconElementList component renders IconElementList using Snapshots 1`] = `
<div
  className={undefined}
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchTap={[Function]}
  style={
    Object {
      "display": "inline-block",
      "muiPrepared": true,
      "position": "relative",
    }
  }>
  <button
    disabled={false}
    href={undefined}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onMouseOut={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    onTouchTap={[Function]}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
        "background": "none",
        "border": 10,
        "boxSizing": "border-box",
        "cursor": "pointer",
        "display": "inline-block",
        "fontFamily": "Roboto, sans-serif",
        "fontSize": 0,
        "fontWeight": "inherit",
        "height": 48,
        "margin": 0,
        "muiPrepared": true,
        "outline": "none",
        "overflow": "visible",
        "padding": 12,
        "position": "relative",
        "textDecoration": "none",
        "transform": "translate(0, 0)",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "verticalAlign": null,
        "width": 48,
      }
    }
    tabIndex={0}
    type="button">
    <div
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}>
      <svg
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "inline-block",
            "fill": "currentColor",
            "height": 24,
            "muiPrepared": true,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none",
            "width": 24,
          }
        }
        viewBox="0 0 24 24">
        <path
          d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z" />
      </svg>
    </div>
  </button>
  <div
    style={
      Object {
        "display": "none",
      }
    } />
</div>
`;
